src/helpers/form.ts

```diff
@@ -0,0 +1,37 @@
+import { emailRegex } from "./regex"
+
+export type SignupErrorCodeType = "auth/email-already-in-use" | "auth/invalid-email" | "auth/weak-password"
+export type SigninErrorCodeType = "auth/invalid-email" | "auth/user-disabled" | "auth/user-not-found" | "auth/wrong-password"
+type errorCodeType = SignupErrorCodeType | SigninErrorCodeType
+
+export type SignupErrorType = {
+  code: SignupErrorCodeType
+} | undefined
+
+export type SigninErrorType = {
+  code: SigninErrorCodeType
+} | undefined
+
+type errorType = SignupErrorType | SigninErrorType | null
```

Comment:
For clarity, we can get rid of these types, and just use the types above. instead of having `arg:SigningErrorType` just use `arg:SigninErrorCodeType|undefined`. Right now the 2 error are a little confusing, and just adds a unnecessary layer of nesting